sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala

```diff
@@ -528,4 +530,65 @@ class HiveScriptTransformationSuite extends BaseScriptTransformationSuite with T
     checkAnswer(query2, identity, Row("\\N,\\N,\\N") :: Nil)
 
   }
+
+  test("SPARK-34879: HiveInspector supports DayTimeIntervalType and YearMonthIntervalType") {
+    assume(TestUtils.testCommandAvailable("/bin/bash"))
+    withTempView("v") {
+      val df = Seq(
+        (Duration.ofDays(1),
+          Duration.ofSeconds(100).plusNanos(123456),
+          Duration.ofSeconds(Long.MaxValue / DateTimeConstants.MICROS_PER_SECOND),
```

Comment:
I would propose to test the corner case `Duration.of(Long.MaxValue, ChronoUnit.MICROS)`

Reply:
> I would propose to test the corner case `Duration.of(Long.MaxValue, ChronoUnit.MICROS)`

Done